Add write to transparency log after building and signing.

jiggoha opened the pull request:
Callouts to a few things proposed in this PR:

* Origin: `transparency.dev/armored-witness/binary_transparency/0`
* Example Claimant Model statement (TAG_NAME is blank in development, should not be in production):
[firmware-log-sequence_applet_log_entry.txt](https://github.com/transparency-dev/armored-witness-applet/files/12163933/firmware-log-sequence_applet_log_entry.txt)

The output of the digest from `sha256sum` includes the filename as well, thinking of stripping that (but would need to add a field to indicate what this is, e.g. trust_applet, trusted_os)?

mhutchinson commented on cloudbuild.yaml:
```diff
@@ -58,12 +58,48 @@ steps:
+      - '-c'
+      - |
+        cp output/trusted_applet.elf . && \
+        echo -e "{\n  digest: $(sha256sum -b trusted_applet.elf),\n  tag_name: ${TAG_NAME},\n  tamago_version: ${_TAMAGO_VERSION},\n}" > output/applet_log_entry
```
You probably only want the digest and not the filename in here?

```suggestion
        echo -e "{\n  digest: $(sha256sum -b trusted_applet.elf | cut -d " " -f1),\n  tag_name: ${TAG_NAME},\n  tamago_version: ${_TAMAGO_VERSION},\n}" > output/applet_log_entry
```

The current output contains the filename `digest: 4b832434ff1d403eb32d8b265647240bdc68e3e22e3323edf91d2907cc08f81b *output/trusted_applet.elf,`

jiggoha replied:
Done.

mhutchinson commented:
This has an output that looks like:
```
{
  digest: 4b832434ff1d403eb32d8b265647240bdc68e3e22e3323edf91d2907cc08f81b,
  tag_name: ,
  tamago_version: 1.20.6,
}
```

As I've started writing the update code, I'm writing the primary Believer consumer for this. This leads to a few questions:

### Question 1: applet or OS?

Is the plan:
 a) to have separate logs for the OS and the applet
 b) to combine them in the same log?

If (b) then we'll need to have an enum so that the update mechanism can determine what this binary is (applet or OS).

### Question 2: which is the latest version?

I'm expecting tag_name to be parseable as a semantic version, e.g. 1.2.3. Is that your expectation?

### Question 3: format

What format is this? I initially thought it was json, but looking more closely I don't see any quotes around the keys and values. If we look at the amory drive log, there's a struct defined in the [api](https://github.com/usbarmory/armory-drive-log/blob/master/api/log_entries.go), and this allows the [leaves](https://github.com/usbarmory/armory-drive-log/blob/master/log/seq/00/00/00/00/01) to be easily parsed. Can we do something similar here?

jiggoha commented:
Thanks for the followup questions.

Question 1: applet or OS?
We discussed face-to-face agreed on a single log, so I've added a "component" field to the Statement.

Question 2: which is the latest version?
Yes. I've added what the Statement looks like with the latest commit for this PR [here](https://github.com/transparency-dev/armored-witness-applet/files/12233617/statement.example.txt).

Question 3: format
Yes, you're right. It should be JSON format. (The formatting was fixed in a later commit, but the earlier example did not reflect this.)

In addition, we agreed face-to-face:

1. to add the Github commit id along with the release tag, since tags are mutable.
2. to add a productionization AI to move the JSON-construction somewhere else more programmatic so it's less brittle.

mhutchinson commented:
This provides what I need to continue with the firmware update component. LGTM, ta!

mhutchinson commented:
See #77 - I found a few problems with this:
 - json needs trailing commas after each key/value pair (except the last one)
 - the sha256_digest is being serialized with hex but should use base64
   - You could pipe it through `xxd -r -p | base64` to do this, but it's looking hacky now! `sha256sum -b output/trusted_applet.elf | cut -d' ' -f1 | xxd -r -p | base64` - yikes! But again, we can change this afterwards to do it via the struct I'm writing in #77.

jiggoha commented:
Fixed the JSON comma issue @mhutchinson mentioned. I tried the `xxd` suggestion as well, but it doesn't look straightforward as I'll need a Cloud Builder image which has that command installed already. Looks like it's gotten to the point that the Jenga blocks are falling. My next immediate TODO is to factor out the JSON construction to code so that we can get the hex/base64 issue sorted.

In the meantime, will finally merge this PR since it's a good checkpoint while @mhutchinson and Al are OOO.
